File: python/p0001_two_sum/solutions/brute_force.py
# Time complexity: O(n^2)
# Space complexity: O(n)

from typing import List


class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        for i, num in enumerate(nums):
            complement = target - num
            for j in range(i + 1, len(nums)):
                if nums[j] == complement:
                    return [i, j]

File: rust/p0001_two_sum/src/brute_force.rs
// Time complexity: O(n^2)
// Space complexity: O(n)

pub struct Solution;

impl Solution {
    pub fn two_sum(nums: Vec<i32>, target: i32) -> Vec<i32> {
        for i in 0..nums.len() - 1 {
            let complement: i32 = target - nums[i];

            for j in i + 1..nums.len() {
                if nums[j] == complement {
                    return vec![i as i32, j as i32];
                }
            }
        }

        unreachable!("This should not happen with valid input");
    }
}
